src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -0,0 +1,118 @@
+
+            var codeFixCollections = Array.Empty<CodeFixCollection>().ToImmutableArray();
+            var codeRefactorings = Array.Empty<CodeRefactoring>().ToImmutableArray();
+
+            var codeFixCollectionsTask = Task.Run(
+                async () => codeFixCollections = await codeFixService.GetFixesAsync(
+                    document, textSpan, includeSuppressionFixes: true, cancellationToken).ConfigureAwait(false));
+            var codeRefactoringsTask = Task.Run(
+                async () => codeRefactorings = await codeRefactoringService.GetRefactoringsAsync(
+                    document, textSpan, cancellationToken).ConfigureAwait(false));
+
+            await Task.WhenAll(codeFixCollectionsTask, codeRefactoringsTask).ConfigureAwait(false);
+            return (codeFixCollections, codeRefactorings);
+        }
+
+        public static CodeAction? GetCodeActionToResolve(string distinctTitle, ImmutableArray<CodeAction> codeActions)
+        {
+            // Searching for the matching code action. We compare against the unique identifier
+            // (e.g. "Suppress or Configure issues.Configure IDExxxx.Warning") instead of the
+            // code action's title (e.g. "Warning") since there's a chance that multiple code
+            // actions may have the same title (e.g. there could be multiple code actions with
+            // the title "Warning" that appear in the code action menu if there are multiple
+            // diagnostics on the same line).
+            foreach (var c in codeActions)
+            {
+                var action = CheckForMatchingAction(c, distinctTitle, currentTitle: "");
+                if (action != null)
+                {
+                    return action;
+                }
+            }
+
+            return null;
+        }
+
+        private static CodeAction? CheckForMatchingAction(CodeAction codeAction, string goalTitle, string currentTitle)
+        {
+            // Adding a delimiter for nested code actions, e.g. 'Suppress or Configure issues.Suppress IDEXXXX.in Source'
+            if (currentTitle != "")
+            {
+                currentTitle += '.';
+            }
+
+            // If the unique identifier of the current code action matches the unique identifier of the code action
+            // we're looking for, return the code action. If not, check to see if one of the current code action's
+            // nested actions may be a match.
+            var updatedTitle = currentTitle + codeAction.Title;
+            if (updatedTitle == goalTitle)
+            {
+                return codeAction;
+            }
+
+            foreach (var nestedAction in codeAction.NestedCodeActions)
+            {
+                var match = CheckForMatchingAction(nestedAction, goalTitle, updatedTitle);
+                if (match != null)
+                {
+                    return match;
+                }
+            }
+
+            return null;
```

Comment:
same concern.

Reply:
I think returning null is valid in this particular method since the caller loops through the top-level code actions to check for a match. So we should definitely find a match in some instance of this method, but not necessarily the current instance (if that makes sense).